cpp/src/IceStorm/Service.cpp

```diff
@@ -228,20 +225,14 @@ ServiceI::start(const string& name, const shared_ptr<Communicator>& communicator
 
             // Determine the set of node id and node proxies.
             //
-            // This is determined by locating all topic manager
-            // replicas, and then working out the node for that
-            // replica.
+            // This is determined by locating all topic manager replicas, and then working out the node for that replica.
             //
-            // We work out the node id by removing the instance
-            // name. The node id must follow.
-            //
-            auto locator = Ice::checkedCast<IceGrid::LocatorPrx>(communicator->getDefaultLocator());
-            assert(locator);
+            // We work out the node id by removing the instance name. The node id must follow.
+            IceGrid::LocatorPrx locator(communicator->getDefaultLocator().value());
             auto query = locator->getLocalQuery();
             auto replicas = query->findAllReplicas(communicator->stringToProxy(instanceName + "/TopicManager"));
 
-            for(Ice::ObjectProxySeq::const_iterator p = replicas.begin(); p != replicas.end(); ++p)
-            for(const auto& replica : replicas)
+            for (const auto& replica : replicas)
```

Comment:
Note there was two loops here, the first one was a leftover of the C++11 port. Its iterator "p" was used in the body, not sure how the tests didn't catch this.